be2-scala/src/main/scala/ch/epfl/pop/model/network/method/message/data/ActionType.scala

```diff
@@ -25,3 +25,9 @@ enum ActionType(val action: String):
   case post_transaction extends ActionType("post_transaction")
   // popcha
   case authenticate extends ActionType("authenticate")
+  // federation actions
+  case init extends ActionType("init")
+  case expect extends ActionType("expect")
+  case challenge_request extends ActionType("challenge_request")
+  case challenge extends ActionType("challenge")
+  case federation_result extends ActionType("federation_result")
```

Comment:
If I am not mistaken, string should match actions from jsonRPC format. "federation_result" -> "result"